File: backend/demo1/src/main/java/com/example/demo/entity/Book.java
package com.example.demo.entity;

import com.example.demo.enums.BookStatus;
import com.example.demo.enums.DocumentType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "books")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @Column(nullable = false)
    String title;

    String author;

    String isbn;

    @Min(value = 1, message = "Invalid year of publication")
    Integer yearOfPublication;

    @Column(nullable = false)
    String category;

    @Min(value = 0, message = "Number of copies can not be negative")
    Integer availableCopies;

    String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    BookStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    DocumentType documentType;

    @OneToMany(mappedBy = "book")
    @JsonIgnore
    private List<BorrowBookRequest> pendingRequests;
}

File: backend/backend-dotnet/Models/Book.cs
using backend_dotnet.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models;

[Table("books")]
public class Book
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public string Title { get; set; } = string.Empty;

    public string? Author { get; set; }

    public string? Isbn { get; set; }

    public int? YearOfPublication { get; set; }

    [Required]
    public BookCategory Category { get; set; }

    public int? AvailableCopies { get; set; }

    public string? ImageUrl { get; set; }

    public string? BookOnlineUrl { get; set; }

    public string? DocumentFileUrl { get; set; }

    [Required]
    public BookStatus Status { get; set; }

    [Required]
    public DocumentType DocumentType { get; set; }

    public ICollection<BorrowBookRequest> PendingRequests { get; set; } = new List<BorrowBookRequest>();

    public ICollection<BookHold> Holds { get; set; } = new List<BookHold>();

    public ICollection<BookReview> Reviews { get; set; } = new List<BookReview>();
}
